Calibration/HcalCalibAlgos/macros/CalibCorr.C

```diff
@@ -0,0 +1,173 @@
+#include <algorithm>
+#include <map>
+#include <vector>
+#include <string>
+#include <iomanip>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+class CalibCorr {
+public :
+  CalibCorr(const std::string& infile, bool debug=false);
+  ~CalibCorr() {}
+
+  float getCorr(int run, unsigned int id);
+private:
+  void                     readCorr(const std::string& infile);
+  std::vector<std::string> splitString(const std::string&);
+  unsigned int getDetIdHE(int ieta, int iphi, int depth);
+  unsigned int getDetId(int subdet, int ieta, int iphi, int depth);
+  unsigned int correctDetId(unsigned int detId);
+
+  static const unsigned int nmax_=10;
+  bool                      debug_;
+  std::map<unsigned int,float> corrFac_[nmax_];
+  std::vector<int>             runlow_;
+};
+
+CalibCorr::CalibCorr(const std::string& infile, bool debug) : debug_(debug) {
+  readCorr(infile);
+}
+
+float CalibCorr::getCorr(int run, unsigned int id) {
+  float cfac(1.0);
+  int ip(-1);
+  for (unsigned int k=0; k<runlow_.size(); ++k) {
+    unsigned int i = runlow_.size()-k-1;
+    if (run >= runlow_[i]) {
+      ip = (int)(i); break;
+    }
+  }
+  if (debug_) std::cout << "Run " << run << " Perdiod " << ip << std::endl;
+  unsigned idx = correctDetId(id);
+  if (ip >= 0) {
+    std::map<unsigned int,float>::iterator itr = corrFac_[ip].find(idx);
+    if (itr != corrFac_[ip].end()) cfac = itr->second;
+  }
+  if (debug_) {
+    int subdet = (idx >> 25) & (0x7);
+    int depth  = (idx >> 20) & (0xF);
+    int zside  = (idx&0x80000)?(1):(-1);
+    int ieta   = (idx >> 10) & (0x1FF);
+    int iphi   = (idx) & (0x3FF);
+    std::cout << "ID " << std::hex << id << std::dec << " (Sub " << subdet 
+	      << " eta " << zside*ieta << " phi " << iphi << " depth " << depth
+	      << ")  Factor " << cfac << std::endl;
+  }
+  return cfac;
+}
+
+void CalibCorr::readCorr(const std::string& infile) {
+
+  std::ifstream fInput(infile.c_str());
```

Comment:
please drop `.c_str()` as there is also a constructor taking directly the `std::string`

Reply:
Did not work

Reply:
It has to work for ROOT 5.26 as well

Reply:
same question as above: why do you need it for ROOT 5.26?